hera_cal/reflections.py

```diff
@@ -225,34 +157,63 @@ def model_reflections(self, dly_range, keys=None, data='clean', edgecut=0, taper
-
-    def write_reflections(self, output_calfits, input_calfits=None, overwrite=False, add_to_history=''):
+
+            # fit for reflection delays and amplitude
+            amp, dly, inds, sig = fit_reflection_delay(self.dfft[k], dly_range, self.delays, return_peak=False)
+
+            # make significance cut
+            if np.max(sig) < ref_sig_cut:
+                continue
+
+            # solve for phase (slowest step)
+            phs = fit_reflection_phase(self.dfft[k], dly_range, self.delays, amp, dly, fthin=fthin, Nphs=Nphs, ifft=True, ifftshift=True)
+
+            # anchor phase to 0 Hz
+            phs = (phs - 2 * np.pi * dly / 1e9 * (self.freqs[0] - zeropad * self.dnu)) % (2 * np.pi)
+
+            # form epsilon term
+            eps = construct_reflection(self.freqs, amp, dly / 1e9, phs, real=False)
+
+            self.ref_eps[rkey] = eps
+            self.ref_amp[rkey] = amp
+            self.ref_phs[rkey] = phs
+            self.ref_dly[rkey] = dly
+            self.ref_significance[rkey] = sig
+            self.ref_flags[rkey] = np.min(flags[k], axis=1, keepdims=True)
+
+        # form gains
+        self.ref_gains = form_gains(self.ref_eps)
+        for a in self.ants:
+            for p in self.pols:
+                k = (a, uvutils.parse_jpolstr(p)) 
+                if k not in self.ref_gains:
+                    self.ref_gains[k] = np.ones((self.Ntimes, self.Nfreqs), dtype=np.complex)
+
+    def write_auto_reflections(self, output_calfits, input_calfits=None, overwrite=False,
+                               write_npz=False, add_to_history=''):
         """
+        Write auto reflection gain terms from self.ref_gains.
+
         Given a filepath to antenna gain calfits file, load the
-        calibration, incorporate reflection term from self.epsilon dictionary
-        and write to output.
+        calibration, incorporate auto-correlation reflection term from the
+        self.ref_gains dictionary and write to file. Optionally, take
+        the values in self.ref_amp, self.ref_phs and self.ref_dly and
+        write their values to an NPZ file with the same path name as output_calfits.
 
         Args:
             output_calfits : str, filepath to write output calfits file
             input_calfits : str, filepath to input calfits file to multiply in with
                 reflection gains.
             overwrite : bool, if True, overwrite output file
+            write_npz : bool, if True, write an NPZ file holding reflection
+                params with the same pathname as output_calfits
             add_to_history: string to add to history of output calfits file
 
         Returns:
-            uvc : HERACal object with new gains
+            uvc : UVCal object with new gains
         """
         # Create reflection gains
-        rgains = _form_gains(self.epsilon)
+        rgains = self.ref_gains
         flags, quals, tquals = None, None, None
 
         if input_calfits is not None:
```

Comment:
Something is weird here. Why would they not match? It seems to me like if the number of times and/or the number of freqs doesn't match, it's an error.